pxr/usd/sdf/testenv/testSdfParsing.py

```diff
@@ -64,6 +64,7 @@ def test_Basic(self):
         # This will mean that your new test runs first and you can spot
         # failures much quicker.
         testFiles = '''
+        217_utf8_identifiers.sdf
         216_bad_variant_in_relocates_path.sdf
         215_bad_variant_in_specializes_path.sdf
         214_bad_variant_in_inherits_path.sdf
```

Comment:
Could we add tests that validate that the text file parser rejects "identifiers" that aren't actually XID_Start/XID_Continue identifiers?  We look for 'identifier's in several places other than prim/property names, and it seems like with this change as-is, we might accept non-identifiers in those places?

Reply:
yeah ok, it will accept unexpected type names, for example, which I added a test for.  I can add back in the extra validation that wsa done in the original modifications I made to the lexer last year, as long as we are aware that we pay a double cost in validation in some cases.

Reply:
Added a few more tests and updated validation rules